src/rewriters/DivModRewriter.h

```diff
@@ -29,17 +31,45 @@ class DivModConfig : public DefaultRewriterConfig {
     std::unordered_map<std::pair<PTRef, PTRef>, DivModPair, PTRefPairHash> divModCache;
     vec<PTRef> definitions;
 
-    const char * divPrefix = ".div";
-    const char * modPrefix = ".mod";
-
     DivModPair freshDivModPair(PTRef dividend, PTRef divisor) {
         std::string id = "_" + std::to_string(dividend.x) + "_" + std::to_string(divisor.x);
-        std::string divName = divPrefix + id;
-        std::string modName = modPrefix + id;
+        std::string divName(divPrefix);
+        divName += id;
+        std::string modName(modPrefix);
+        modName += id;
         return {logic.mkIntVar(divName.c_str()), logic.mkIntVar(modName.c_str())};
     }
 
+    static opensmt::pair<PTRef, PTRef> getDividendAndDivisor(std::string_view const name,
+                                                             std::string_view const prefix) {
+        std::string dividendNumberStr;
+        std::string divisorNumberStr;
+        bool parsingDividend = true;
+        for (auto it = name.begin() + prefix.size() + 1; it != name.end(); ++it) {
+            if (parsingDividend and '0' <= *it and *it <= '9') {
+                dividendNumberStr += *it;
+            } else if (not parsingDividend and '0' <= *it and *it <= '9') {
+                divisorNumberStr += *it;
+            } else if (*it == '_') {
+                assert([](bool parsingDividend, std::string_view const name) {
+                    if (not parsingDividend) {
+                        throw OsmtInternalException("Parse error in auxiliary variable symbol: " + std::string(name));
+                        return false;
+                    } else {
+                        return true;
+                    }
+                }(parsingDividend, name));
```

Comment:
Seems like unnecessary complicated piece of code. Can we just do
```suggestion
                    if (not parsingDividend) { throw OsmtInternalException("Parse error in auxiliary variable symbol: " + std::string(name)); }
```

Reply:
Looks ok to me.